src/agrep.c

```diff
@@ -449,6 +453,11 @@ tre_agrep_handle_file(const char *filename)
 		  printf("%.*s", record_len, record);
 		}
 	    }
+            /* exit after max_count match(es) */
+            if((max_count > 0)&&(count > (max_count-1))) {
+              if(count_matches) break;/* to get the result */
+              else exit(0);/* to leave the stream "as is" */
```

Comment:
Exiting here produces the wrong result when multiple inputs are provided and at least one of them has `max_count` or more matching lines. The correct thing to do is break in all cases. Moreover, we're reading from a file descriptor, not a stream, and since we read a variable amount of data (but at least 10 kB) at a time, there is no guarantee that the input will be left in any particular state when we reach the match limit.

Reply:
Yes, I added the ugly `else exit(0);` as a quick fix for my use case.

There were some trouble either when piping or using multiple files, but I can't recall what problems exactly, sorry m(_ _)m